### ISSUE TITLE
Index does not always have same format for field value

### ISSUE THREAD

**Issue by bjarnef** [Shazwazza/Examine#84]
Hi @Shazwazza 

I have a multi-site solution, currently running on Umbraco version 7.5.8
The sites contains some reviews in a listview, site 1 contains ~10K review nodes and site 2 contains ~7K review nodes.

I have a Examine index for these review nodes for each site, where one of the indexes fields is a true/false property. In the db the value is 0 or 1 for this property for all nodes, but in the index it is sometimes 0/1 or other times it is false/true.

I noticed it because a custom review property returned 500 internal error, because Covert.ToBoolean fails on "0" og "1", but works on values like "true", "TRUE", "false", "FALSE".

so in my code I have changed from this:
````
bool showComment = r.Fields.ContainsKey("reviewShowComment") ? Convert.ToBoolean(r.Fields["reviewShowComment"]) == true : false; 
````

to this:
```` 
bool showComment = r.Fields.ContainsKey("reviewShowComment") ? (r.Fields["reviewShowComment"].ToLower() == "true" || r.Fields["reviewShowComment"] == "1" ? true : false) : false;
```` 

IndexSet of Site 1
![javaw_2017-05-11_15-17-03](https://cloud.githubusercontent.com/assets/2919859/26028332/aa2a93ee-381e-11e7-89f9-2bde5f40b0c0.png)

IndexSet of Site 2

![javaw_2017-05-11_15-19-22](https://cloud.githubusercontent.com/assets/2919859/26028326/87d59a6e-381e-11e7-95d1-93e9138aa319.png)

Have you noticed this behaviour?

In another site on Umbraco 7.5.13 I have in the ExternalIndex a blogDate property which most of the time contains a valid datetime in the index, but also contains values that not are valid datetimes.

![javaw_2017-05-13_20-58-34](https://cloud.githubusercontent.com/assets/2919859/26028374/6c13f892-381f-11e7-8d5b-8122018158c3.png)

**Comment by Shazwazza** [comment]
Is it maybe because you have old/stale data in your index? What happens when you rebuild the index, do these inconsistent values remain?

Also note that in Umbraco core we have an extension method on `object`: `TryConvertTo<T>` which is ultra powerful and far more robust than using `Convert.ToBoolean` it will take into account 0, 1, true, False, etc... so you can do:

```csharp
var showComment  = false;
var booleanConvert = r.Fields["reviewShowComment"].TryConvertTo<bool>();
if (booleanConvert.Success) {
   //TODO: do stuff if it could be converted
   showComment  = booleanConvert.Result;
}
```

**Comment by bjarnef** [comment]
Thanks @Shazwazza 
I have noticed the TryConvertTo method before, but didn't thought about it is a Umbraco core method.
Does is handle null values if e.g. `r.Fields["reviewShowComment"]` is null.

I have changed my code to the following:

```
bool showComment = false;
if(r.Fields.ContainsKey("reviewShowComment"))
{
	var showCommentConvert = r.Fields["reviewShowComment"].TryConvertTo<bool>();
	if (showCommentConvert.Success)
	{
		showComment = showCommentConvert.Result;
	}
}
```

I have actually tried to rebuild the Examine index, but I still have entries with 0 and 1 values. Is it indended to be 0/1 or false/true?

Funny enough Site 1 have most entries with false/true (and a few 0/1 entries) and the opposite for Site 2.

The other example with `blogDate` is a quite new site running on Umbraco Cloud, where I wonder how the index got the values `00` and `20`.
